pkg/controller/master/setting/auto_rotate_rke2_certs.go

```diff
@@ -0,0 +1,214 @@
+package setting
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	provisioningv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+const (
+	defaultReconcilAutoRotateRKE2CertsSettingDuration = time.Hour * 24 // 1 day
+)
+
+func (h *Handler) syncAutoRotateRKE2Certs(setting *harvesterv1.Setting) error {
+	logrus.WithFields(logrus.Fields{
+		"name":  setting.Name,
+		"value": setting.Value,
+	}).Info("Processing setting")
+
+	autoRotateRKE2Certs := &settings.AutoRotateRKE2Certs{}
+	if err := json.Unmarshal([]byte(setting.Value), autoRotateRKE2Certs); err != nil {
+		logrus.WithFields(logrus.Fields{
+			"name":  setting.Name,
+			"value": setting.Value,
+		}).WithError(err).Error("failed to unmarshal setting value")
+		return err
+	}
+	if !autoRotateRKE2Certs.Enable {
+		return nil
+	}
+
+	kubernetesIP, err := h.getKubernetesIP()
+	if err != nil {
+		return err
+	}
+	if kubernetesIP == "" {
+		logrus.WithFields(logrus.Fields{
+			"name":              setting.Name,
+			"service.namespace": metav1.NamespaceDefault,
+			"service.name":      "kubernetes",
+			"reconcileAfter":    defaultReconcilAutoRotateRKE2CertsSettingDuration,
+		}).Warn("cluster ip is empty, reconcile setting again")
+		h.settingController.EnqueueAfter(setting.Name, defaultReconcilAutoRotateRKE2CertsSettingDuration)
+		return nil
+	}
```

Comment:
this is unlikely to happen, we should may be just let this error and reconcile at normal backoff interval

Reply:
Updated it. Thank you.